### PR TITLE
feat: SEC-824 Add CodeQL scan feature.

### PR DESCRIPTION
**Pull request by brianlamCC**
Testing Github's CodeQL scanning with our public repositories.

[SEC-824](https://cloudconformity.atlassian.net/browse/SEC-824).

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (brianlamCC) [inline .github/workflows/codeql-analysis.yml:1]: This file is the default configuration provided by Github - I have made no changes.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file. 
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (iaindurham) [inline .github/workflows/codeql-analysis.yml:57]: ♻️  You can probably remove this step as JS doesn't require any building

**Review by tkhawari — APPROVED** [review]
looks good

**Review by iaindurham — APPROVED** [review]
I have also turned on the passing of these checks as a requirement before any PR can be merged
image

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (brianlamCC) [reply .github/workflows/codeql-analysis.yml:57]: Thanks @iaindurham

**Pull request merged**